front/lib/api/workspace.ts

```diff
@@ -37,6 +37,8 @@ export async function getWorkspaceInfos(
     allowedDomain: workspace.allowedDomain,
     role: "none",
     segmentation: workspace.segmentation,
+    // Flags are not needed from outside of the workspace.
+    flags: [],
```

Comment:
This one doesn't feel perfect IMO, because we are kind of introducing a lighter version of Workspace and it insn't reflected in the type system. Is this function guaranteed to be called only for a specific use-case ?

Reply:
Happy to | null the type. We already do that for other elements of it when retrieved for other purposes than the main auth flow 👍

Reply:
👍  But then the function that return workspace type must return either:
`WorkspaceType & {flags: null}` or `WorkspaceType & {flags: string[]}` right ?

Reply:
Yes we can add that type guard for convenience in the code at the point of usage 👍

Reply:
Ok fine ! FWIW my preference would be to have clear type signatures for functions that don't lie

Reply:
Otherwise, if you call a function that may return null flags, you need to handle the case where you didn't get the flags when you needed them, so you have a runtime error (that would have been prevented with proper types)